Bumping electron to 18.3.6 triggers typescript errors

#2921 depends on this

### Actual Behavior

* Set electron to `^18.3.6` in `package.json`
* rm -fr package-lock.json node_modules
* npm i
* npm run lint:typescript:nofix

```
…/src/store/prefs.js
  223:5   warning  Caution: `Vue` also has a named export `set`. Check if you meant to write `import {set} from 'vue'` instead  import/no-named-as-default
```
(repeated 5 times over 2 files)
```
* npm run test:unit:jest
```
Test suite failed to run

    src/utils/fetch.ts:117:13 - error TS2322: Type 'boolean | Agent | undefined' is not assignable to type 'Agent'.
      Type 'undefined' is not assignable to type 'Agent'.

    117             agent = options.agent(parsedURL);
```

--- a/package.json
+++ b/package.json
@@ -111,7 +111,7 @@
     "@typescript-eslint/parser": "^5.33.0",
     "@vue/test-utils": "^1.1.2",
     "babel-core": "^7.0.0-bridge.0",
-    "electron": "^18.0.3",
+    "electron": "^18.3.6",
     "electron-builder": "^23.3.3",
     "electron-devtools-installer": "^3.2.0",
     "electron-notarize": "^1.2.1",

